src/components/Identity/syncDeclaredIds.js

```diff
@@ -0,0 +1,52 @@
+import { sha256Buffer, normalizeCustomerIds, bufferToHex } from "./util";
+import { crc32 } from "../../utils";
+import { COOKIE_NAMES } from "./constants";
+
+const { CUSTOMER_ID_HASH } = COOKIE_NAMES;
+
+const makeServerCall = (payload, lifecycle, network) => {
+  return lifecycle.onBeforeDataCollection(payload).then(() => {
+    return network.sendRequest(payload, payload.expectsResponse);
+  });
+};
+
+export default (
+  customerIds,
+  event,
+  payload,
+  cookie,
+  lifecycle,
+  network,
+  optIn
+) => {
+  const normalizedCustomerIds = normalizeCustomerIds(customerIds);
+  const idNames = Object.keys(normalizedCustomerIds);
+  const idsToHash = idNames.filter(idName => customerIds[idName].hash);
+  const hashedIds = idsToHash.map(id =>
```

Comment:
Maybe call this `idHashPromises` or something to indicate that they are promises and not the actual hashed Ids...